lib/MusicBrainz/Server/Entity/Relationship.pm

```diff
@@ -279,9 +279,9 @@ around TO_JSON => sub {
         entity1_credit  => $self->entity1_credit,
         entity0_id      => $self->entity0_id,
         entity1_id      => $self->entity1_id,
-        id              => $self->id + 0,
-        linkOrder       => $self->link_order + 0,
-        linkTypeID      => $link->type_id + 0,
+        id              => ($self->id // 0) + 0,
+        linkOrder       => ($self->link_order // 0) + 0,
+        linkTypeID      => ($link->type_id // 0) + 0,
```

Comment:
The `linkOrder` change seems fine, though it might be better for us to add `default => 0` to the Moose attribute instead (since that's also the column default at the database level).

For the other two, I'd rather not silence any warnings those are causing without identifying why the `id` and `linkTypeID` are undefined there. I'm guessing it might be from relationships that haven't been created yet or have incomplete info from edit data, but don't know for certain.

If there's a reason for them to be undefined, I do wonder if it'd be safer to output them as `undef` instead of 0 in case we have any checks that assume anything that's a number is a valid ID, e.g. https://github.com/metabrainz/musicbrainz-server/blob/87f0ab5/root/static/scripts/edit/MB/edit.js#L157

Reply:
Agreed, silencing warnings here without identifying when these happen was just bad.

Note: even `link_order` might have a reason to be undefined: https://github.com/metabrainz/musicbrainz-server/blob/v-2020-02-18/root/edit/details/add_relationship.tt#L8